src/api_manager/proto/server_config.proto

```diff
@@ -233,4 +233,14 @@ message Experimental {
   bool disable_log_status = 1;
   // Configure response to JSON translator option for JsonPrintOptions:
   bool always_print_primitive_fields = 2;
+
+  // Defines the request header field whose value will be used to be used
+  // as the real client address.
+  string client_real_ip_header = 3;
```

Comment:
I think we only deal with XFF,  not need to support getting ip from other headers.   so just adding a config field
xff_postion_for_client_ip

Reply:
Changed

Reply:
Separated RealClientIPFromHeaderConfig message